### ISSUE TITLE
protocol: grpc: Add new function ListProcesses()

### ISSUE THREAD

**Issue by sboeuf** [kata-containers/agent#73]
@jodh-intel pointed out to me that we need a way to ask the agent about the processes running on one of its container. I think a simple way would be to only rely on `ContainerID` as the only argument of this function.
About what it should return, I am not sure what we should expect. IMO, we have two options:
- Either we return an array of bytes, being the output of `ps ...` command being run on the agent
- Or we agree on a global structure like this
```go
type ProcessList struct {
        PID   string
        Ppid string
        CPU  string
        Cmd  string
        ...
}
```
that we would return, leaving the formatting of those data up to the runtime.

**Comment by sboeuf** [comment]
cc @sameo @bergwolf @laijs @gnawux @WeiZhang555 @jodh-intel

**Comment by jodh-intel** [comment]
cc @devimc.

**Comment by WeiZhang555** [comment]
Is this used for `kata-runtime ps` command？

**Comment by bergwolf** [comment]
I'd vote for a global structure, so that `kata runtime ps` can replace a process's pid with corresponding shim's pid when returning.

**Comment by sboeuf** [comment]
@WeiZhang555 yes that's the idea.
@bergwolf thanks for your feedback.

**Comment by WeiZhang555** [comment]
I think a global structure is better than raw stream of ‘ps’, it can make keeping backward compatibility easier in future

**Comment by sboeuf** [comment]
@WeiZhang555 cool, that's would be my choice too.

**Comment by jodh-intel** [comment]
Implemented on #193.
